### PR TITLE
[DOXIA-534] Migrate logging to Sl4j

### PR DESCRIPTION
**Pull request by slachiewicz**
Please check. 

Current build fails with cirr error - due to interfaces change.

### CODE REVIEW

File: doxia-modules/doxia-module-fo/pom.xml [hunk 1]
```diff
@@ -78,7 +72,7 @@ under the License.
     <dependency>
       <groupId>commons-collections</groupId>
       <artifactId>commons-collections</artifactId>
-      <version>3.2.1</version>
+      <version>3.2.2</version>
```
Comment (michael-o) [inline doxia-modules/doxia-module-fo/pom.xml]: This should be a seperate issue if not necesary for this PR.

File: pom.xml [hunk 2]
```diff
@@ -185,15 +180,33 @@ under the License.
...
+        <scope>test</scope>
       </dependency>
     </dependencies>
   </dependencyManagement>
   <dependencies>
     <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>4.11</version>
+      <version>4.12</version>
```
Comment (michael-o) [inline pom.xml]: Same as commons-collections

File: doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java [hunk 3]
```diff
@@ -1322,12 +1322,12 @@ private SinkEventAttributeSet getGraphicsAttributes( String logo )
         }
         catch ( IOException e )
         {
-            getLog().debug( e );
+            getLog().debug( e.getMessage() );
```
Comment (michael-o) [inline doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java]: Leave it as `e`, SFL4J and backends will do the rest for us.

**Comment by michael-o** [comment]
@hboutemy Looks good to me. Tests pending...

**Comment by hboutemy** [comment]
I'm not in favor of removing doxia-logging: deprecating and modifying it to use slf4j would be more compatible
I know there are not so much Doxia extensions in the wild: there is at least asciidoc I know of. I didn't check, but would be useful to not break this one at least

**Comment by michael-o** [comment]
Why not deprecate in 1.8 and remove with 2.0?

**Comment by hboutemy** [comment]
+1

**Comment by michael-o** [comment]
@slachiewicz Can you spawn another ticket and create a new PR? We will leave this one open.

**Comment by slachiewicz** [comment]
There is almost no usage of logging in Doxia, so any extention needs only to remove one call.

File: doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java (hunk 3 above)
Comment (slachiewicz) [reply doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java]: there is no constructor with Exception, i will change to e.getMessage(), e

File: pom.xml (hunk 2 above)
Comment (slachiewicz) [reply pom.xml]: ok, will be diffret pull - only with dependences

File: doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java (hunk 3 above)
Comment (hazendaz) [reply doxia-modules/doxia-module-fo/src/main/java/org/apache/maven/doxia/module/fo/FoAggregateSink.java]: @slachiewicz @michael-o Proper code here would be ```getLog().debug("", e)```.  That assumes you want the full stack trace and not just the message.

**Comment by slachiewicz** [comment]
New version with deprecation in #8

**Comment by abelsromero** [comment]
> I'm not in favor of removing doxia-logging: deprecating and modifying it to use slf4j would be more compatible
> I know there are not so much Doxia extensions in the wild: there is at least asciidoc I know of. I didn't check, but would be useful to not break this one at least

I mantain the asciidoctor extension and I am fine with removing logging-api. In fact we have problems sharing code between the our normal Mojo and the doxia parser because one uses "maven.plugin.logging.Log" and the other "maven.doxia.logging.Log". Moving doxia to Sl4j won't fix it 100%, but will be one step closer.

**Pull request closed without merging**